src/filerepository.js

```diff
@@ -0,0 +1,455 @@
+/**
+ * @license Copyright (c) 2003-2017, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+/**
+ * @module upload/filerepository
+ */
+
+import Plugin from '@ckeditor/ckeditor5-core/src/plugin';
+
+import CKEditorError from '@ckeditor/ckeditor5-utils/src/ckeditorerror';
+import ObservableMixin from '@ckeditor/ckeditor5-utils/src/observablemixin';
+import Collection from '@ckeditor/ckeditor5-utils/src/collection';
+import mix from '@ckeditor/ckeditor5-utils/src/mix';
+import log from '@ckeditor/ckeditor5-utils/src/log';
+
+import FileReader from './filereader.js';
+
+import uid from '@ckeditor/ckeditor5-utils/src/uid.js';
+
+/**
+ * FileRepository plugin.
+ *
+ * @extends module:core/plugin~Plugin
+ */
+export default class FileRepository extends Plugin {
+	/**
+	 * @inheritDoc
+	 */
+	static get pluginName() {
+		return 'fileRepository';
```

Comment:
@Reinmar: according to the recent convention should it be "upload/fileRepository"? I must say I don't like these plugins namespaces. They make names much longer does not let overwrite plugins nicely in the future and might be a problem if we will try to move plugins between packages (note that from the end-developer perspective it will be the only thing which will change).